src/rpcmining.cpp

```diff
@@ -415,8 +416,14 @@ Value getblocktemplate(const Array& params, bool fHelp)
 
     LOCK(cs_main);
 
+    // Wallet is required because we support coinbasetxn
+    if (pwalletMain == NULL) {
+        throw JSONRPCError(RPC_METHOD_NOT_FOUND, "Method not found (disabled)");
+    }
```

Comment:
Originally I had it fall back to using `coinbaseval`, and it would make sense to do so. But this caused issues with the existing caching when the wallet *was* enabled and anyone first fetched using `coinbaseval` - a subsequent `coinbasetxn` would use the block template containing the dummy key, and any mining rewards would be lost.

A few possible alternatives:
- Have two cached block templates - one for `coinbasetxn` with a real key, and one for `coinbaseval` with a dummy.
- Cache a boolean recording which type was last requested, and use it to invalidate the block template cache.
- Mock `CReserveKey` if there is no wallet.
- Remove the cache (not a good idea).

If we do allow usage of this API without a wallet, it needs to throw an error on `coinbasetxn`.